File: app/src/main/java/com/app/gymbuzz/fragments/GymDetailFragment.java
package com.app.gymbuzz.fragments;

import android.os.Bundle;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.LinearSnapHelper;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.app.gymbuzz.R;
import com.app.gymbuzz.entities.GymDetailModel;
import com.app.gymbuzz.fragments.abstracts.BaseFragment;
import com.app.gymbuzz.ui.binders.GymImageBinder;
import com.app.gymbuzz.ui.views.AnyTextView;
import com.app.gymbuzz.ui.views.CustomRecyclerView;
import com.app.gymbuzz.ui.views.TitleBar;
import com.bumptech.glide.Glide;
import com.nostra13.universalimageloader.core.ImageLoader;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import de.hdodenhof.circleimageview.CircleImageView;

public class GymDetailFragment extends BaseFragment {
    @BindView(R.id.rvGymImages)
    CustomRecyclerView rvGymImages;
    Unbinder unbinder;
    @BindView(R.id.civProfilePic)
    CircleImageView civProfilePic;
    @BindView(R.id.txtGymName)
    AnyTextView txtGymName;
    @BindView(R.id.txtGymAddress)
    AnyTextView txtGymAddress;
    @BindView(R.id.txtStatus)
    AnyTextView txtStatus;
    @BindView(R.id.txtAboutUs)
    AnyTextView txtAboutUs;
    private GymDetailModel gymDetailModel;

    public static GymDetailFragment newInstance(GymDetailModel gymDetailModel) {
        Bundle args = new Bundle();

        GymDetailFragment fragment = new GymDetailFragment();
        fragment.setArguments(args);
        fragment.setGymDetailModel(gymDetailModel);
        return fragment;
    }

    public void setGymDetailModel(GymDetailModel gymDetailModel) {
        this.gymDetailModel = gymDetailModel;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
        }

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_gym_detail, container, false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        bindGymDate();
    }

    private void bindGymDate() {
        if (gymDetailModel != null) {
            ImageLoader.getInstance().displayImage(gymDetailModel.getLogopath(), civProfilePic);
            txtGymName.setText(gymDetailModel.getGymname() + "");
            txtGymAddress.setText(gymDetailModel.getAddress() + " " + gymDetailModel.getCity());
            txtAboutUs.setText(gymDetailModel.getDescription() + "");
            rvGymImages.bindRecyclerView(new GymImageBinder(), gymDetailModel.getGymimages(),
                    new LinearLayoutManager(getDockActivity(), LinearLayoutManager.HORIZONTAL, false), new DefaultItemAnimator());
            new LinearSnapHelper().attachToRecyclerView(rvGymImages);
        }
    }

    @Override
    public void setTitleBar(TitleBar titleBar) {
        super.setTitleBar(titleBar);
        titleBar.hideButtons();
        titleBar.showBackButton();
        titleBar.setSubHeading(getString(R.string.gym_detail));
        titleBar.showNotificationButton(0);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: app/src/main/java/com/app/gymbuzz/fragments/GymDetailFragment.kt
package com.app.gymbuzz.fragments

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.app.gymbuzz.R
import com.app.gymbuzz.fragments.abstracts.BaseFragment
import com.app.gymbuzz.interfaces.ItemClickListener
import com.app.gymbuzz.ui.adapters.RecyclerViewAdapter
import com.app.gymbuzz.ui.binders.BinderGymDetail
import com.app.gymbuzz.ui.views.TitleBar
import kotlinx.android.synthetic.main.fragment_gym_detail.*
import kotlinx.android.synthetic.main.gym_detail_item.*

/**
 * Created on 5/23/2018.
 */


class GymDetailFragment:BaseFragment(),ItemClickListener{

    lateinit var gymImageList: MutableList<String>

    private var rvAdapter: RecyclerViewAdapter<String>? = null

    companion object {
        val Tag: String = "GymDetailFragment"
        fun newInstance(): GymDetailFragment {
            val fragment = GymDetailFragment()
            return fragment
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_gym_detail,container,false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        bindData()

    }

    fun bindData(){

        gymImageList = mutableListOf<String>()
        gymImageList.add(0,"drawable://" + R.drawable.gym_floor_1)
        gymImageList.add(1,"drawable://" + R.drawable.gym_floor_2)
        gymImageList.add(2,"drawable://" + R.drawable.gym_floor_1)

        rvAdapter = RecyclerViewAdapter(gymImageList, BinderGymDetail(this), dockActivity, R.layout.gym_detail_item)
        rvGymImages.setLayoutManager(LinearLayoutManager(mainActivity,LinearLayoutManager.HORIZONTAL,false))
        rvGymImages.setAdapter(rvAdapter)
        rvAdapter?.notifyDataSetChanged()


    }

    override fun setTitleBar(titleBar: TitleBar?) {
        super.setTitleBar(titleBar)

        titleBar?.hideButtons()
        titleBar?.showBackButton()
        titleBar?.setSubHeading(getString(R.string.gym_detail))

    }

    override fun itemClicked(imagePath: String?, position: Int) {
        dockActivity.replaceDockableFragment(ImageViewFragment.newInstance(), "ImageViewFragment")
    }

}